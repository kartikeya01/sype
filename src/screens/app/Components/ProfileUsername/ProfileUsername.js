import React from 'react';
import {StyleSheet, Text} from 'react-native';
import PropTypes from 'prop-types';

export default class ProfileUsername extends React.Component {
  render() {
    const username = this.props.username;
    /** Made this its own class if we ever want to go back and utalize a library to style it,
     * we would just target this. Though we can remove this class and just move everything abck to ProfileBanner.
     */
    return <Text style={styles.usernameFontSize}>{username}</Text>;
  }
}

ProfileUsername.propTypes = {
  username: PropTypes.string.isRequired,
};

const styles = StyleSheet.create({
  usernameFontSize: {
    fontSize: 25,
  },
});
