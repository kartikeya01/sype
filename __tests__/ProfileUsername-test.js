import 'react-native';
import React from 'react';
import ProfileUsername from '@screens/app/Components/ProfileUsername/ProfileUsername';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import checkPropTypes from 'check-prop-types';
var assert = require('assert'); /** NODE.js assert function */

/** ================ HELPER FUNCTIONS ================ */

function testAllProperPropsPassed(username) {
  it('Username value is the ProfileUsername prop value', () => {
    const wrapper = shallow(<ProfileUsername username={username} />);
    const instance = wrapper.instance();
    const usernameProp = instance.props.username;
    expect(username).toEqual(usernameProp);
  });
}

/** ================ SNAPSHOT TEST ================ */

test('Profile Username Snapshot Test', () => {
  const tree = renderer.create(<ProfileUsername username={''} />).toJSON();
  expect(tree).toMatchSnapshot();
});

/** ================ UNIT TESTS ================ */
/**  -- To do, localize + refactor code. */

describe('Profile Username Unit Testing', () => {
  const stringUsernameTest = 'bsefa';
  testAllProperPropsPassed(stringUsernameTest);

  it('Throws failed propType erorr for not providing any props', () => {
    let result = checkPropTypes(
      ProfileUsername.propTypes,
      {},
      'prop',
      ProfileUsername,
    );
    console.log(result);
    assert(
      result.includes(
        'Failed prop type: The prop `username` is marked as required',
      ),
    );
  });

  it('Throws failed propType erorr for not providing username', () => {
    let result = checkPropTypes(
      ProfileUsername.propTypes,
      {},
      'prop',
      ProfileUsername,
    );
    console.log(result);
    assert(
      result.includes(
        'Failed prop type: The prop `username` is marked as required',
      ),
    );
  });

  it('Throws Failed propType Error for providing username value of undefined and not required value', () => {
    let result = checkPropTypes(
      ProfileUsername.propTypes,
      {username: undefined},
      'prop',
      ProfileUsername,
    );
    console.log(result);
    assert(
      result.includes(
        'Failed prop type: The prop `username` is marked as required in `function ProfileUsername',
      ) && result.includes('but its value is `undefined`.'),
    );
  });

  it('Throws Failed propType Error for providing username value of true and not string', () => {
    let result = checkPropTypes(
      ProfileUsername.propTypes,
      {username: true},
      'prop',
      ProfileUsername,
    );
    console.log(result);
    assert(
      result.includes(
        'Failed prop type: Invalid prop `username` of type `boolean` supplied to `function ProfileUsername',
      ) && result.includes('expected `string`.'),
    );
  });

  it('Throws Failed propType Error for providing username value of false and not string', () => {
    let result = checkPropTypes(
      ProfileUsername.propTypes,
      {username: false},
      'prop',
      ProfileUsername,
    );
    console.log(result);
    assert(
      result.includes(
        'Failed prop type: Invalid prop `username` of type `boolean` supplied to `function ProfileUsername',
      ) && result.includes('expected `string`.'),
    );
  });

  it('Throws Failed propType Error for providing username value of integer number and not string', () => {
    let result = checkPropTypes(
      ProfileUsername.propTypes,
      {username: 12},
      'prop',
      ProfileUsername,
    );
    console.log(result);
    assert(
      result.includes(
        'Failed prop type: Invalid prop `username` of type `number` supplied to `function ProfileUsername',
      ) && result.includes('expected `string`.'),
    );
  });

  it('Throws Failed propType Error for providing username value of double number and not string', () => {
    let result = checkPropTypes(
      ProfileUsername.propTypes,
      {username: 12.1},
      'prop',
      ProfileUsername,
    );
    console.log(result);
    assert(
      result.includes(
        'Failed prop type: Invalid prop `username` of type `number` supplied to `function ProfileUsername',
      ) && result.includes('expected `string`.'),
    );
  });

  it('Throws Failed propType Error for providing username value of array and not string', () => {
    let result = checkPropTypes(
      ProfileUsername.propTypes,
      {username: []},
      'prop',
      ProfileUsername,
    );
    console.log(result);
    assert(
      result.includes(
        'Failed prop type: Invalid prop `username` of type `array` supplied to `function ProfileUsername',
      ) && result.includes('expected `string`.'),
    );
  });

  it('Throws Failed propType Error for providing username value of object and not string', () => {
    let result = checkPropTypes(
      ProfileUsername.propTypes,
      {username: {}},
      'prop',
      ProfileUsername,
    );
    console.log(result);
    assert(
      result.includes(
        'Failed prop type: Invalid prop `username` of type `object` supplied to `function ProfileUsername',
      ) && result.includes('expected `string`.'),
    );
  });
});
