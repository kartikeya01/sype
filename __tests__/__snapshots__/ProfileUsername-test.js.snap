// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Username Snapshot Test 1`] = `
<Text
  style={
    Object {
      "fontSize": 25,
    }
  }
>
  
</Text>
`;
